base/namedtuple.jl

```diff
@@ -273,6 +273,7 @@ values(nt::NamedTuple) = Tuple(nt)
 haskey(nt::NamedTuple, key::Union{Integer, Symbol}) = isdefined(nt, key)
 get(nt::NamedTuple, key::Union{Integer, Symbol}, default) = haskey(nt, key) ? getfield(nt, key) : default
 get(f::Callable, nt::NamedTuple, key::Union{Integer, Symbol}) = haskey(nt, key) ? getfield(nt, key) : f()
+tail(t::NamedTuple{names}) where names = NamedTuple{tail(names)}(tail(values(t)))
```

Comment:
I think `NamedTuple{tail(names)}(t)` will work, and is a bit better since it will preserve field types.